### PR TITLE
added support for xsiam

### PR DESCRIPTION
**Pull request by moishce**
## Status
Ready/In Progress/In Hold (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/46625

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Unit Test or Example Code
- [ ] Changelog entry

### CODE REVIEW

**Comment by glicht** [comment]
Need to also update release notes and main README.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 [1]: https://pypi.org/project/demisto-py/#history
 
+## 3.1.2
+* Added auth_id to support `XSIAM` instances.
```
Comment (Itay4) [inline CHANGELOG.md:8]:
```suggestion
* Add support to Cortex XSIAM.
```

File: README.md [hunk 2]
```diff
@@ -25,6 +25,7 @@ as the following environment variables (env variables will be used if parameters
 * DEMISTO_API_KEY
 * DEMISTO_USERNAME
 * DEMISTO_PASSWORD
+* DEMISTO_AUTH_ID (for XSIAM, If is set it means we should use the XSIAM api)
```
Comment (Itay4) [inline README.md:28]: i think the env var should be named `XSIAM_AUTH_ID`

Comment (Itay4) [inline README.md:28]:
```suggestion
* DEMISTO_AUTH_ID (for Cortex XSIAM only. If set, Cortex XSIAM API will be used)
```

File: README.md [hunk 3]
```diff
@@ -41,7 +42,20 @@ api_instance = demisto_client.configure(base_url=host, api_key=api_key)
 
 ```
 
-**Alternatively, you can login with username and password:**
+**on the XSIAM we nead to add the auth_id**
```
Comment (Itay4) [inline README.md:45]:
```suggestion
**For Cortex XSIAM, we need to set the auth_id**
```

File: README.md [hunk 4]
```diff
@@ -41,7 +42,20 @@ api_instance = demisto_client.configure(base_url=host, api_key=api_key)
...
+```python
+import demisto_client
+
+# Also possible to set env variables: DEMISTO_API_KEY and DEMISTO_BASE_URL
```
Comment (Itay4) [inline README.md:49]: add note here about the possible env var

File: demisto_client/__init__.py [hunk 5]
```diff
@@ -71,9 +77,12 @@ def configure(base_url=None, api_key=None, verify_ssl=None, proxy=None, username
                 connection_pool_maxsize = int(connection_pool_maxsize)
     configuration = Configuration()
     configuration.api_key['Authorization'] = api_key
-    configuration.host = base_url or os.getenv('DEMISTO_BASE_URL', None)
+    configuration.host = base_url
     if isinstance(configuration.host, str):
         configuration.host = configuration.host.rstrip('/')
+    if auth_id:
+        configuration.api_key['x-xdr-auth-id'] = auth_id
+        configuration.host = f'{configuration.host}/xsoar'
```
Comment (Itay4) [inline demisto_client/__init__.py:85]: use `os.path.join` to avoid issues with trailing slash

**Review by Itay4 — APPROVED** [review]

**Review by Itay4 — APPROVED** [review]

**Pull request merged**